Data/Aeson/Compat.hs

```diff
@@ -0,0 +1,16 @@
+{-# LANGUAGE CPP #-}
+
+module Data.Aeson.Compat
+  (
+    fromStrict
+  ) where
+
+import qualified Data.ByteString      as S
+import qualified Data.ByteString.Lazy as L
+
+fromStrict :: S.ByteString -> L.ByteString
```

Comment:
This can definitely be inlined!

Reply:
Won't the compiler decide to inline the function automatically?

Reply:
I hope so! But I'm not sure...

Reply:
If we're talking about GHC, the documentation says this:

>GHC (with -O, as always) tries to inline (or “unfold”) functions/values that are “small enough,” thus avoiding the call overhead and possibly exposing other more-wonderful optimisations. GHC has a set of heuristics, tuned over a long period of time using many benchmarks, that decide when it is beneficial to inline a function at its call site. The heuristics are designed to inline functions when it appears to be beneficial to do so, but without incurring excessive code bloat. If a function looks too big, it won’t be inlined, and functions larger than a certain size will not even have their definition exported in the interface file. Some of the thresholds that govern these heuristic decisions can be changed using flags, see -f*: platform-independent flags.

>Normally GHC will do a reasonable job of deciding by itself when it is a good idea to inline a function. However, sometimes you might want to override the default behaviour. For example, if you have a key function that is important to inline because it leads to further optimisations, but GHC judges it to be too big to inline.

https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html?inline-and-noinline-pragmas#inline-and-noinline-pragmas.